arduino/src/arduino.cpp

```diff
@@ -0,0 +1,285 @@
+ * @brief Construct a new Arduino:: Arduino object
+ * 
+ */
+Arduino::Arduino(std::string portName, unsigned int buadRate, unsigned int timeout) : portName(portName), buadRate(buadRate), timeout(timeout)
+{
+}
+/**
+ * @brief getter, will return an array of files that the class uses
+ * to output of the sensors
+ * 
+ * @return std::vector<std::string> an array of string 
+ */
+std::vector<std::string> Arduino::getFiles()
+{
+    return this->files;
+}
+/**
+ * @brief getter for the timeout of the serial port
+ * 
+ * @return int the time to abandon the serial communication
+ */
+int Arduino::getTimout()
+{
+    return this->timeout;
+}
+/**
+ * @brief setter of the timeout
+ * 
+ * @param timeout int the time to abandon the serial communication
+ */
+void Arduino::setTimeout(int timeout)
+{
+    this->timeout = timeout;
+}
+/**
+ * @brief check the connection to the arduino and report.
+ */
+Arduino::ERROR_ARDUINO Arduino::checkConnection()
+{
+    if (!this->isOpen)
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    char helper = this->serial.openDevice(portName.c_str(), buadRate);
+    if (helper != SUCCESS)
+    {
+        serial.closeDevice();
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    }
+    serial.closeDevice();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief check the connection for debugging
+ */
+void Arduino::checkConnectionToConsole()
+{
+    if (checkConnection() != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        std::cout << "there was a problem with the connection" << std::endl;
+    }
+    std::cout << "the connection is successful!-" << std::endl;
+}
+/**
+ * @brief add file to the files array
+ * @param file add a file to the array of files
+ */
+
+void Arduino::addFile(std::string file)
+{
+    this->files.push_back(file);
+}
+/*
+ * @brief will calculate the sum of a string
+ * I forgot that I found this function on the web so I made 
+ * a simple version of this using std::string
+ * this one will probably be used in the arduino
+ * @param message char array pointer 
+ * @return int the sum of the chars ascii values.
+ */
+int Arduino::checkSum(char *message)
+{
+    int sum = 0;
+    char *p = &message[0];
+    while (*p != NULLCHAR)
+    {
+        sum += *message;
+        ++p;
+    }
+    return sum;
+}
+/**
+ * @brief open the serial class with the members of this class
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::openSerial()
+{
+    int err = this->serial.openDevice(this->portName.c_str(), this->buadRate);
+    if (err != SUCCESS)
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    this->isOpen = true;
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/*
+ * @brief send a serial commend to the arduino and confirm
+ *  that the message has been received 
+ * 
+ * @param message  the commend to be sent, for now 3 is the prefix 
+ * of the commends and the 2 digits behind it are the commend itself
```

Comment:
try to read this sentence and tell me if you can understand it

Reply:
fixed